tests/test_cli/test_provision.py

```diff
@@ -39,6 +39,15 @@ def mock_use_lock():
         yield mocked_use_lock
 
 
+@pytest.fixture(autouse=True)
+def mock_use_remote_state():
```

Comment:
Should this mock also be present in `test_destroy.py` since we use the context manager there as well? Or should it be inside `test_cli/conftest.py` so that all tests use mocked version of `use_remote_state` context manager.

Reply:
Thanks Shubham. I didn't mock it in destroy (or higher up) as the tests didn't seem to call for it. 

I've also mocked it differently (replacing the upload and download functionality in RemoteStateManager) in the `RemoteStateManager` tests, which is why I avoided moving the mocked `use_remote_state` to a higher `conftest`. I'd be happy to move it up if you think there's value in it.